index.js

```diff
@@ -18,85 +19,114 @@ program.parse(process.argv);
 
 const options = program.opts();
 
+const diagnoseWebSocket = (endpoint, origin) => {
+  return new Promise((resolve, reject) => {
+    const client = new WebSocketClient();
+    client.on('connectFailed', (error) => {
+      identifyProblem(`Connect Failed: ${error.toString()}`);
+      if (!error.code) {
+        error.code = "ENOTFOUND";
+      }
+      reject(error);
+    });
+    
+    client.on('connect', (connection) => {
+      connection.on('error', (error) => {
+        identifyProblem(`Connection Error: ${error.toString()}`);
+        reject(error);
+      });
+      connection.on('close', () => {
+        resolve();
+      });
+    });
+      
+    client.connect(endpoint, undefined, origin);
+})};
+
+let hasIdentifiedProblem = false;
+let hasIdentifiedCorsProblem = false;
+const identifyProblem = (...problemDescription) => {
+  hasIdentifiedProblem = true;
+  console.log("⚠️ ", ...problemDescription);
+};
+const identifyCorsProblem = (...problemDescription) => {
+  identifyProblem(...problemDescription);
+  hasIdentifiedCorsProblem = true;
+}
 (async () => {
```

Comment:
OOh these could probably stay inside the async function, because they are about the state of the execution, `diagnoseWebSocket` is a helper that resolves to the resolve so it makes sense outside of the function :) 
```suggestion
(async () => {
  let hasIdentifiedProblem = false;
  let hasIdentifiedCorsProblem = false;
  const identifyProblem = (...problemDescription) => {
    hasIdentifiedProblem = true;
    console.log("⚠️ ", ...problemDescription);
  };
  const identifyCorsProblem = (...problemDescription) => {
    identifyProblem(...problemDescription);
    hasIdentifiedCorsProblem = true;
  }
```